generators/csharp/sdk/src/endpoint/snippets/SnippetJsonGenerator.ts

```diff
@@ -0,0 +1,91 @@
+import { FernGeneratorExec } from "@fern-fern/generator-exec-sdk";
+import { Endpoint } from "@fern-fern/generator-exec-sdk/api";
+import { ExampleEndpointCall, HttpEndpoint } from "@fern-fern/ir-sdk/api";
+import { SdkGeneratorContext } from "../../SdkGeneratorContext";
+import { GrpcEndpointGenerator } from "../grpc/GrpcEndpointGenerator";
+import { HttpEndpointGenerator } from "../http/HttpEndpointGenerator";
+import urlJoin from "url-join";
+import { RootClientGenerator } from "../../root-client/RootClientGenerator";
+import { SingleEndpointSnippet } from "./EndpointSnippetsGenerator";
+
+export class SnippetJsonGenerator {
+    private readonly context: SdkGeneratorContext;
+    private readonly rootClientGenerator: RootClientGenerator;
+    constructor({ context }: { context: SdkGeneratorContext }) {
+        this.context = context;
+        this.rootClientGenerator = new RootClientGenerator(context);
+    }
+
+    public generate(): FernGeneratorExec.Snippets | undefined {
+        const rootClientSnippet = this.rootClientGenerator
+            .generateExampleClientInstantiationSnippet()
+            .toFormattedSnippet({
+                allNamespaceSegments: this.context.getAllNamespaceSegments(),
+                allTypeClassReferences: this.context.getAllTypeClassReferences(),
+                rootNamespace: this.context.getNamespace(),
+                customConfig: this.context.customConfig
+            });
+        const rootClientImportList = rootClientSnippet.imports?.split("\n") ?? [];
+
+        function getCsharpSnippet(endpointSnippet: SingleEndpointSnippet): string {
+            const snippetImportList = endpointSnippet.imports?.split("\n") ?? [];
+            const uniqueOrderedImports = Array.from(new Set([...rootClientImportList, ...snippetImportList]))
+                .filter((importString) => importString !== "")
+                .sort();
+            return `${uniqueOrderedImports.join("\n")}\n\nvar client = ${rootClientSnippet.body}${
+                endpointSnippet.endpointCall
+            }`;
+        }
```

Comment:
I would have expected this to be an additional top-level `private` function. If we need access to the `rootClientSnippet` defined above, we can expose another parameter (rather than relying on the closure).

Otherwise, it's a little tough to follow when the function is defined within the `generate` function body.